### DISCUSSION TITLE
Localization of info.plist for IOS

### DISCUSSION THREAD

**Discussion by FM1973** [dotnet/maui#15928] [Q&A] [answered]
Hey there.

Has anyone been able to localize the info.plist file for IOS?
I´ve seen some tutorials for Xamarin Forms, but that doens´t seem to work in .net Maui.

I´ve created the necessary folders for the InfoPlist.strings files.
I.e. en.lproj and de.lproj inside Platforms/IOS/Resources

The .strings files follow the rule "key"="localized-value";

But I can´t set the build action to BundleResource.

**Comment by FM1973** [comment, marked as answer]
I found a solution on reddit...

Since BundleResource does not exist as a build action for .net maui, one can do the following in the project file:

`<ItemGroup Condition="'$(TargetFramework)' == 'net7.0-ios'">
    <MauiAsset Include="Platforms\iOS\Resources\de.lproj\InfoPlist.strings" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
  </ItemGroup>`

In this case my default language is english and I set the text for english directly in the info.plist file.
For "german" I followed the tutorials and created a file InfoPlist.strings in Platforms\iOS\Resources\de.lproj.

This does work, just keep in mind you have to set the regional settings right.

**Reply by andyzukunft** [reply]
Hey @FM1973,

this might be a little late however I would ask follow up questions on how you got this to work.

I created a `Platforms\iOS\Resources\de.lproj\InfoPlist.strings` file. 

I tried to use 
```xml
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">
    <BundleResource Include="Platforms\iOS\Resources\de.lproj\InfoPlist.strings" />
</ItemGroup>
```
and
```xml
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">
    <MauiAsset Include="Platforms\iOS\Resources\de.lproj\InfoPlist.strings" LogicalName="%(RecursiveDir)%(Filename)%(Extension)" />
</ItemGroup>
```

So far both option do not provide a tarnslation. I would like to translate some permission informations strings.

Example Info.plist:
```
<key>NSPhotoLibraryUsageDescription</key>
<string>Reason description</string>
```

Example de.lproj\InfoPlist.strings:
```
/* German Language File */
"NSPhotoLibraryUsageDescription" = "Hier steht ein Text mit der Begründung";
```

Do you translate something else?

**Reply by Brionvega** [reply]
Hi Andy!

I encounter the same issue than you, while following each step, it never shows the language on iPhones UI when asking for sharing location or camera. 

Have you managed to make it work?

**Reply by hugebug4ever** [reply]
following settings works for me
```
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'ios'">
        <BundleResource Include="Platforms\iOS\Resources\en.lproj\InfoPlist.strings" LogicalName="en.lproj/%(Filename)%(Extension)" /> 
</ItemGroup>
```

**Reply by jonathanantoine** [reply]
For thos having the same issue than me : if you put the 'InfoPlist' fils in the same place than "LogicalName", then you will have a "duplication" issue so create a nesting (useless) folder to fix it. I choose to name it 'Platform" but it can be anything :

![image](https://github.com/user-attachments/assets/37e84069-9c73-4cf3-b7f5-3999f88823b2)
